async.lua

```diff
@@ -0,0 +1,23 @@
+areas = rawget("_G", "areas") or {}
```

Comment:
Didn't test this; turns out it's wrong.

```suggestion
areas = rawget(_G, "areas") or {}
```